src/ray/protobuf/experimental/autoscaler.proto

```diff
@@ -24,22 +24,40 @@ import "src/ray/protobuf/experimental/instance_manager.proto";
 // Following fields represents the Cluster Resources autoscaler interested
 // in.
 
-// Represents an anti-affinity constraint. A bundle with this constraint
-// can't be allocated to a node that has a label with the same name and
-// value. This is used to implement placement group anti-affinity.
+// Represents a placement constraint to specify how an associated resource
+// request should be placed with other entity, e.g.
+//  1. Other resource requests (e.g. gang scheduling)
+//  2. Nodes (e.g. specific instance type)
 //
-// For placement group, the label_name is "_PG" (reserved),
-// and the label_value is the placement group id.
-message AntiAffinityConstraint {
-  string label_name = 1;
-  string label_value = 2;
-  // If true, the label will be created on the node
-  // where the request with this constraint is scheduled.
-  bool create_label_on_schedule = 3;
-}
-
+// For now, it only supports constraints generated from placment groups:
+//  - STRICT_SPREAD: bundles will have the same ANTI_AFFINITY constraint,
+//      [type=ANTI_AFFINITY, key=_PG, value=pg_id]
+//  - STRICT_PACK:
+//      [type=AFFINITY, key=_PG, value=pg_id]
+//
+// While speculative, this should be flexible enough to support other types of
+// constraints, where the key-value pair will be used to limit placement. e.g.
+//  - Placed on a cloud provider: [type=AFFINITY, key="Cloud", value=cloud_provider]
+//  - Certain instance type: [type=AFFINITY, key="Instance", value=instance_type]
+//  - With customized resources: [type=AFFINITY, key="MY_RESOURCE", value=""]
+//
+// Users of this (i.e. autoscaler) should be free enough to evaluate the constraint
+// and make decisions based on it.
 message PlacementConstraint {
```

Comment:
hmm the intention is to make PlacementConstraint contains one of affinity constraints, in the future it will be have other constraints lke

```
message PlacementConstraint {
  AntiAffinityConstraint anti_affinity = 1;
  AnyPlacementConstraints any_constraints = 2;
  ...
}
```

the new change seemingly make it harder.